### ISSUE TITLE
LGTM.com warning

### ISSUE THREAD

**Issue by DimitriPapadopoulos** [lamyj/dicomifier#46]
Here are the warnings raised by LGTM.com:
https://lgtm.com/projects/g/lamyj/dicomifier/?severity=warning

**Comment by DimitriPapadopoulos** [comment]
> Variable defined multiple times
> This assignment to `PixelValueTransformation` is unnecessary as it is redefined [here](https://github.com/lamyj/dicomifier/blob/c7a49bbcdd6d7876946e17a19950bed64b1101ae/src/python/dicomifier/bruker_to_dicom/modules/image.py#L283) before this value is used.

https://lgtm.com/projects/g/lamyj/dicomifier/snapshot/0b81c150d5e2022f77e0ff3654c3c8b0af5a6ca6/files/src/python/dicomifier/bruker_to_dicom/modules/image.py?sort=name&dir=ASC&mode=heatmap#x91f8a5a7933c7ae1:1

Fixed by discarding the first occurrence and keeping the last one. The last one is more recent, introduced by 8188ea8.

**Comment by DimitriPapadopoulos** [comment]
> Variable defined multiple times
> This assignment to '`ascconv`' is unnecessary as it is redefined [here](https://github.com/lamyj/dicomifier/blob/c7a49bbcdd6d7876946e17a19950bed64b1101ae/src/python/dicomifier/dicom_to_nifti/siemens.py#L44) before this value is used.

https://lgtm.com/projects/g/lamyj/dicomifier/snapshot/0b81c150d5e2022f77e0ff3654c3c8b0af5a6ca6/files/src/python/dicomifier/dicom_to_nifti/siemens.py#xba698c2ae01b84b7:1

Fixed by discarding the first occurrence and keeping the last one. I'm not certain what the initial intent was, but at least this fix does not change the current behaviour.

**Comment by DimitriPapadopoulos** [comment]
> Variable defined multiple times
> This assignment to '`x`' is unnecessary as it is redefined [here](https://github.com/lamyj/dicomifier/blob/c7a49bbcdd6d7876946e17a19950bed64b1101ae/src/python/dicomifier/dicom_to_nifti/stacks.py#L393) before this value is used.
> This assignment to '`y`' is unnecessary as it is redefined [here](https://github.com/lamyj/dicomifier/blob/c7a49bbcdd6d7876946e17a19950bed64b1101ae/src/python/dicomifier/dicom_to_nifti/stacks.py#L393) before this value is used.
> This assignment to '`z`' is unnecessary as it is redefined [here](https://github.com/lamyj/dicomifier/blob/c7a49bbcdd6d7876946e17a19950bed64b1101ae/src/python/dicomifier/dicom_to_nifti/stacks.py#L393) before this value is used.

I'm not sure how to fix this warning. I can of course discard the first occurrence, but looking at the code,  I'm wondering whether a deeper change is needed. The current code looks like this:
```python
    if odil.registry.ImageComments not in data_set:
        if b_value_element != 0:
            [...]
            return None
        else:
            x,y,z = [0,0,0]
            image_comments = b"b=0(0,0,0)"
    else:
        image_comments = data_set[odil.registry.ImageComments][0]

    [...]
    try:
        b_value_comment, x, y, z = [float(x) for x in match.groups()]
    [...]

    return image_comments
```
I'm wondering whether the intent was:
```python
    if odil.registry.ImageComments not in data_set:
        if b_value_element != 0:
            [...]
            return None
        else:
            image_comments = b"b=0(0,0,0)"
    else:
        image_comments = data_set[odil.registry.ImageComments][0]

        # Shouldn't all this code be indented to be executed in the "else:" branch?
        # It doesn't look like this need to be executed if `image_comments = b"b=0(0,0,0)"`.
        [...]
        try:
            b_value_comment, x, y, z = [float(x) for x in match.groups()]
        [...]

    return image_comments
```

**Comment by DimitriPapadopoulos** [comment]
> Implicit string concatenation in a list
> Implicit string concatenation. Maybe missing a comma?

https://lgtm.com/projects/g/lamyj/dicomifier/snapshot/0b81c150d5e2022f77e0ff3654c3c8b0af5a6ca6/files/src/python/dicomifier/bruker_to_dicom/convert.py#x770dad99c71da875:1

Not really an issue in my opinion, indenting makes the intent clear. However, fixing LGTM.com false positive when possible at no real cost looks like a good idea. Since the concatenation is deliberate, I have added `+` to join the strings, as suggested by https://lgtm.com/rules/2084600455/.

**Comment by DimitriPapadopoulos** [comment]
Yes, but is it worth parsing `image_comments` if it is `b"b=0(0,0,0)"`? Doesn't seem so.

**Comment by lamyj** [comment]
It felt more logical to me when I wrote it to check and normalize the content and then parse. I don't have strong feelings on this; moreover, this is rather experimental code: Canon MRIs are a pretty rare beast (last I heard, less than 10 are deployed worldwide), and this was reverse-engineered from a single data set.

**Comment by DimitriPapadopoulos** [comment]
Putting everything under `else:` make sense because when the function sets `image_comments`, it doesn't need to parse it. Also it will help have a single exit point in the function:
```python
    if odil.registry.ImageComments not in data_set:
        if b_value_element != 0:
            [...]
            image_comments = None  # remove the previous "return"
        else:
            image_comments = b"b=0(0,0,0)"
    else:
        image_comments = data_set[odil.registry.ImageComments][0]

        [...]
        try:
            b_value_comment, x, y, z = [float(x) for x in match.groups()]
        [...]

    return image_comments  # single exit point

```

**Comment by lamyj** [comment]
This makes sense indeed. Can you integrate this in #47 before I merge it?

**Comment by DimitriPapadopoulos** [comment]
Changed. However, I was too optimistic about the single exit point. There are other `return None` in this function, and removing them would involve adding lots of `if`'s. The code is pretty good as it is.

I could revert the indent under `else:` but I still believe this part of the code does belong to the `else:`  branch.

**Comment by lamyj** [comment]
I'll merge as is; when/if I get more Canon data, I'll have a look at this again.

Thanks for your work on this.
